traveler-ios-ui/TravelerKitUI/ViewControllers/BookablePurchaseSelectionViewController.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  BookablePurchaseSelectionViewController.swift
+//  TravelerKitUI
+//
+//  Created by Ben Ruan on 2019-08-14.
+//  Copyright © 2019 GuestLogix. All rights reserved.
+//
+
+import UIKit
+import TravelerKit
+
+protocol BookablePurchaseSelectionViewControllerDelegate: class {
+
+}
+
+class BookablePurchaseSelectionViewController: UIViewController {
+    @IBOutlet weak var priceLabel: UILabel!
+    @IBOutlet weak var button: UIButton!
+
+    var errorContext: ErrorContext?
+    var bookingContext: BookingContext?
+    weak var delegate: BookablePurchaseSelectionViewControllerDelegate?
+
+    /// TEMP
+
+    private var passes: [Pass]?
+
+    /// END TEMP
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        bookingContext?.addObserver(self)
+        priceLabel.text = bookingContext?.product.price.localizedDescriptionInBaseCurrency
+    }
+
+    deinit {
+        bookingContext?.removeObserver(self)
+    }
+
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        switch (segue.identifier, segue.destination) {
+        case (_, let vc as BookableDetailsViewController):
+            vc.errorContext = errorContext
+            vc.bookingContext = bookingContext
+        case (_, let vc as BookableConfirmationViewController):
+            vc.modalPresentationStyle = .custom
+            vc.modalPresentationCapturesStatusBarAppearance = true
+            vc.passes = passes
+            vc.product = bookingContext?.product
+            vc.delegate = self.delegate as? BookableConfirmationViewControllerDelegate
```

Comment:
We are unsure here that this delegate actually conforms to `BookableConfirmationViewControllerDelegate`

User delegation proxying instead